### PR TITLE
Adding lfric_reader

### PR DESCRIPTION
**Pull request by tinyendian**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there. (I am currently the only maintainer)
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tinyendian** [comment]
@conda-forge-admin, please ping team

**Comment by tinyendian** [comment]
Reade for review, thanks

**Comment by tinyendian** [comment]
@conda-forge-admin, please ping team

**Comment by bjlittle** [comment]
@ocefpaf Could I call in a favour and ask for a merge to unblock this PR? 😉 

I'm collaborating with colleagues at NIWA, and they'd greatly appreciate having their package on `conda-forge` 👍

File: recipes/lfric_reader/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cd src/cxx
+cmake CMakeLists.txt \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DBUILD_SHARED_LIBS=ON \
+      -DBUILD_TESTING=OFF
```
Comment (xhochy) [inline recipes/lfric_reader/build.sh:7]: We have some common CMake arguments, please use them.
```suggestion
cmake ${CMAKE_ARGS} CMakeLists.txt \
      -DCMAKE_BUILD_TYPE=Release \
      -DBUILD_SHARED_LIBS=ON \
      -DBUILD_TESTING=OFF
```

File: recipes/lfric_reader/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - {{ compiler('cxx') }}
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libxext') }}     # [linux]
+    - cmake
+    - boost
+    - eigen
```
Comment (xhochy) [inline recipes/lfric_reader/meta.yaml:27]: These probably belong into `host`

**Comment by tinyendian** [comment]
@xhochy thanks a lot for the review - I implemented all suggested changes. I also added an additional CMake flag that specifies Python version explicitly, after the second package build during the automated GitHub Actions check on Linux suddenly broke, with CMake being unable to find Python.

File: recipes/lfric_reader/build.sh (hunk 1 above)
Comment (tinyendian) [reply recipes/lfric_reader/build.sh]: Implemented in commit below, thanks

File: recipes/lfric_reader/meta.yaml (hunk 2 above)
Comment (tinyendian) [reply recipes/lfric_reader/meta.yaml]: Implemented in commit below, thanks

**Comment by ocefpaf** [comment]
LGTM. @xhochy got here for a review first so I'll let him merge it if he is OK with the changes.

**Pull request merged**